src/main/java/ir/sahab/uncaughtexceptionrule/UncaughtException.java

```diff
@@ -0,0 +1,32 @@
+package ir.sahab.uncaughtexceptionrule;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Wrapper around all exceptions that are occurred during unit test execution.
+ *
+ * @see UncaughtExceptionBase#afterHandle()
+ */
+class UncaughtException extends Exception {
+    static final long serialVersionUID = -7188848759039620762L;
+    private final List<Throwable> exceptions;
+
+    public UncaughtException(String message, List<Throwable> exceptions) {
+        super(message);
+        this.exceptions = exceptions;
+    }
+
+    public List<Throwable> getExceptions() {
+        return exceptions;
+    }
+
+    @Override
+    public String getMessage() {
+        final String messages = exceptions.stream()
+                .map(Throwable::getMessage)
```

Comment:
You forgot to change this to:

.map(ex -> ex.getClass().getSimpleName() + ": " + ex.getMessage())

Reply:
Sorry, I forgot that.